packages/core/src/compose/script-injection.ts

```diff
@@ -0,0 +1,38 @@
+import { ScriptInjection, sectionToLabels } from '@preevy/common'
+import { ComposeModel, ComposeService } from './model'
+
+export const addScript = (model: ComposeModel, service: string, { id, ...script } :
+  {id: string} & ScriptInjection):ComposeModel => {
+  const { services } = model
+  if (!services || !(service in services)) {
+    return model
+  }
+  const serviceDef = services[service]
+  return {
+    ...model,
+    services: {
+      ...model.services,
+      [service]: {
+        ...serviceDef,
+        labels: {
+          ...serviceDef.labels,
+          ...sectionToLabels(`preevy.inject_script.${id}`, script),
+        },
+      },
+    },
+  }
+}
+
+export const scriptInjector = (id : string, script: ScriptInjection) => {
+  const injectScript = (model:ComposeModel, service:string) => addScript(model, service, { id, ...script })
+  const injectAll = (_serviceName:string, _def: ComposeService) => true
+  return {
+    inject: (model: ComposeModel, serviceFilter = injectAll) =>
```

Comment:
I created it initially as a single function, the reason for creating the nested function is for currying the service (id, params) from the injection context (model, filter), otherwise it doesn't provide much value compared to using addScript (which is used externally) directly.

The reason for returning an object and not a single function was for better communicating the action.